packages/core/react/src/WalletProvider.tsx

```diff
@@ -273,7 +286,7 @@ export const WalletProvider: FC<WalletProviderProps> = ({
     return (
         <WalletContext.Provider
             value={{
-                wallets,
+                wallets: readyWallets,
```

Comment:
I'm wondering if we could extend the `Wallet` interface by adding a `ready` boolean attribute. This way the consumers of the WalletContext will have a choice to whether show only ready wallets or something else. Let's say I have a Dapp that supports just a few wallets and I want to show the install status (installed/not installed), I agree that this wouldn't be the main use case, but I think it's nice to have that level of flexibility.

If there was such an attribute for `Wallet`, we could potentially update the `ready` field for all the wallets in the `useEffect` that's currently filtering, that way consumers will be able to do what they prefer with the wallets but have the `ready ` status of all the wallets available.

Reply:
Yeah, this is what I'm unhappy with about the design. I don't want to modify the `Wallet` interface or mutate the instances of it, but I think something like you're talking about would be good. Extending the `Wallet` interface in the React context could work.